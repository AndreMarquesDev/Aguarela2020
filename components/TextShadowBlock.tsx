import React, { FC } from 'react';
import { useWindowSize, mobileBreakpoint, desktopBreakpoint } from '../utils/useWindowSize';
import Button from './Button';

interface TextShadowBlockProps {
    title1: string;
    title2: string;
    title3: string;
    titleMobile: string;
    textBlock1: string;
    textBlock2: string;
    textBlock3: string;
    hasButton?: boolean;
}

const TextShadowBlock: FC<TextShadowBlockProps> = ({ title1, title2, title3, titleMobile, textBlock1, textBlock2, textBlock3, hasButton }) => {
    const windowSize = useWindowSize();
    const isDesktop = windowSize.width > desktopBreakpoint;
    const isMobile = windowSize.width < mobileBreakpoint;

    return (
        <>
            <section className="container">
                <div className="wrapper genericMargins">
                    <div className="titleBlock">
                        {isMobile ? (
                            <p>{titleMobile}</p>
                        ) : (
                            <>
                                <strong>{title1}</strong>
                                <strong>{title2}</strong>
                                <strong>{title3}</strong>
                            </>
                        )}
                    </div>
                    <div className="textBlock">
                        <p
                            // eslint-disable-next-line react/no-danger
                            dangerouslySetInnerHTML={{
                                __html: textBlock1,
                            }}
                            className="bodyText bodyText--white"
                        />
                        <p className="bodyText bodyText--white">{textBlock2}</p>
                        <p className="bodyText bodyText--white">{textBlock3}</p>
                        {hasButton && <Button isUppercased alignLeft={isDesktop} externalLink="mailto:catarina@aguareladigital.pt">catarina@aguareladigital.pt</Button>}
                    </div>
                </div>
            </section>

            <style jsx>
                {`
                    @import './styles/_vars.scss';

                    .container {
                        background: $purple;

                        .wrapper {
                            display: flex;
                            justify-content: space-between;
                            align-items: center;
                            overflow: hidden;

                            @include desktop {
                                flex-direction: column;
                                justify-content: center;
                            }
                        }

                        .titleBlock {
                            width: 35%;

                            @include desktop {
                                width: 100%;
                                margin-bottom: 50rem;
                            }

                            strong {
                                display: block;
                                @include fontXXL($blue, uppercase, 900);
                                letter-spacing: 30rem;
                                text-align: right;
                                -webkit-filter: drop-shadow(20rem -10rem 0 $pink);
                                filter: drop-shadow(20rem -10rem 0 $pink);
                                background: $blue;
                                -webkit-background-clip: text;
                                -webkit-text-fill-color: transparent;
                                text-align: center;
                            }

                            p {
                                @include fontL($blue, uppercase, 900);
                                letter-spacing: 5rem;
                                text-align: center;
                                -webkit-filter: drop-shadow(5rem -2.5rem 0 $pink);
                                filter: drop-shadow(5rem -2.5rem 0 $pink);
                            }
                        }

                        .textBlock {
                            width: 60%;

                            p {
                                text-align: left;
                            }

                            @include desktop {
                                width: 80%;

                                p {
                                    text-align: center;
                                }
                            }

                            @include tablet {
                                width: 100%;
                            }
                        }
                    }
                `}
            </style>
        </>
    );
};

export default TextShadowBlock;
