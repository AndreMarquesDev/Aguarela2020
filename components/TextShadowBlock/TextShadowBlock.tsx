import React, { FC } from 'react';
import { useWindowSize, Breakpoint } from '../../utils/useWindowSize';
import Button from '../Button/Button';

export interface TextShadowBlockProps {
    title1: string;
    title2: string;
    title3: string;
    titleMobile: string;
    textBlock1: string;
    textBlock2: string;
    textBlock3: string;
    hasButton?: boolean;
}

export const textBlock1DataTestId = 'textShadowBlock_textBlock1';
export const buttonText = 'catarina@aguareladigital.com';
export const mailtoValue = 'mailto:catarina@aguareladigital.com';

const TextShadowBlock: FC<TextShadowBlockProps> = ({
    title1,
    title2,
    title3,
    titleMobile,
    textBlock1,
    textBlock2,
    textBlock3,
    hasButton,
}) => {
    const windowSize = useWindowSize();
    const isDesktop = windowSize.width > Breakpoint.Desktop;
    const isMobile = windowSize.width < Breakpoint.Mobile;

    return (
        <>
            <section className="container">
                <div className="wrapper genericMargins">
                    <div className="titleBlock">
                        {isMobile ? (
                            <p>{titleMobile}</p>
                        ) : (
                            <>
                                <strong>{title1}</strong>
                                <strong>{title2}</strong>
                                <strong>{title3}</strong>
                            </>
                        )}
                    </div>
                    <div className="textBlock">
                        <p
                            // eslint-disable-next-line react/no-danger
                            dangerouslySetInnerHTML={{
                                __html: textBlock1,
                            }}
                            className="bodyText bodyText--white"
                            data-testid={textBlock1DataTestId}
                        />
                        <p className="bodyText bodyText--white">{textBlock2}</p>
                        <p className="bodyText bodyText--white">{textBlock3}</p>
                        {hasButton && (
                            <Button isUppercased alignLeft={isDesktop} externalLink={mailtoValue}>
                                {buttonText}
                            </Button>
                        )}
                    </div>
                </div>
            </section>

            <style jsx>
                {`
                    @import './styles/_vars.scss';

                    .container {
                        background: $purple;

                        .wrapper {
                            display: flex;
                            justify-content: space-between;
                            align-items: center;
                            overflow: hidden;

                            @include desktop {
                                flex-direction: column;
                                justify-content: center;
                            }
                        }

                        .titleBlock {
                            width: 35%;

                            @include desktop {
                                width: 100%;
                                margin-bottom: 50rem;
                            }

                            strong {
                                display: block;
                                @include fontXXL($blue, uppercase, 900);
                                letter-spacing: 30rem;
                                text-align: right;
                                -webkit-filter: drop-shadow(20rem -10rem 0 $pink);
                                filter: drop-shadow(20rem -10rem 0 $pink);
                                background: $blue;
                                -webkit-background-clip: text;
                                -webkit-text-fill-color: transparent;
                                text-align: center;
                            }

                            p {
                                @include fontL($blue, uppercase, 900);
                                letter-spacing: 5rem;
                                text-align: center;
                                -webkit-filter: drop-shadow(5rem -2.5rem 0 $pink);
                                filter: drop-shadow(5rem -2.5rem 0 $pink);
                            }
                        }

                        .textBlock {
                            width: 60%;

                            p {
                                text-align: left;
                            }

                            @include desktop {
                                width: 80%;

                                p {
                                    text-align: center;
                                }
                            }

                            @include tablet {
                                width: 100%;
                            }
                        }
                    }
                `}
            </style>
        </>
    );
};

export default TextShadowBlock;
