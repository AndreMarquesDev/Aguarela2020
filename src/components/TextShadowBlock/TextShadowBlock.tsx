import React from 'react';
import { textBlock1DataTestId } from '../../utils/dataTestIds';
import { aguarelaDigitalEmail } from '../../utils/urls';
import { useWindowSize, Breakpoint } from '../../utils/useWindowSize';
import { Button } from '../Button/Button';
import { desktopMediaQuery, tabletMediaQuery } from '../../styles/mediaQueries';

export interface TextShadowBlockProps {
    title1: string;
    title2: string;
    title3: string;
    titleMobile: string;
    textBlock1: string;
    textBlock2: string;
    textBlock3: string;
    dataTestId: string;
    hasButton?: boolean;
}

export const TextShadowBlock = ({
    title1,
    title2,
    title3,
    titleMobile,
    textBlock1,
    textBlock2,
    textBlock3,
    dataTestId,
    hasButton,
}: TextShadowBlockProps): JSX.Element => {
    const windowSize = useWindowSize();
    const isDesktop = windowSize.width > Breakpoint.Desktop;
    const isMobile = windowSize.width < Breakpoint.Mobile;

    return (
        <>
            <section className="container" data-testid={dataTestId}>
                <div className="wrapper genericMargins">
                    <div className="titleBlock">
                        {isMobile ? (
                            <p className="fontL">{titleMobile}</p>
                        ) : (
                            <>
                                <strong className="fontXXL">{title1}</strong>
                                <strong className="fontXXL">{title2}</strong>
                                <strong className="fontXXL">{title3}</strong>
                            </>
                        )}
                    </div>
                    <div className="textBlock">
                        <p
                            dangerouslySetInnerHTML={{
                                __html: textBlock1,
                            }}
                            className="bodyText bodyText--white"
                            data-testid={textBlock1DataTestId}
                        />
                        <p className="bodyText bodyText--white">{textBlock2}</p>
                        <p className="bodyText bodyText--white">{textBlock3}</p>
                        {hasButton && (
                            <Button
                                isLowercased
                                alignLeft={isDesktop}
                                externalLink={`mailto:${aguarelaDigitalEmail}`}
                            >
                                {aguarelaDigitalEmail}
                            </Button>
                        )}
                    </div>
                </div>
            </section>

            <style jsx>
                {`
                    .container {
                        background: var(--purple);

                        .wrapper {
                            display: flex;
                            justify-content: space-between;
                            align-items: center;
                            overflow: hidden;

                            @media (${desktopMediaQuery}) {
                                flex-direction: column;
                                justify-content: center;
                            }
                        }

                        .titleBlock {
                            width: 35%;

                            @media (${desktopMediaQuery}) {
                                width: 100%;
                                margin-bottom: 50rem;
                            }

                            strong {
                                display: block;
                                color: var(--blue);
                                text-transform: uppercase;
                                font-weight: 900;
                                letter-spacing: 30rem;
                                text-align: right;
                                -webkit-filter: drop-shadow(20rem -10rem 0 var(--pink));
                                filter: drop-shadow(20rem -10rem 0 var(--pink));
                                background: var(--blue);
                                -webkit-background-clip: text;
                                -webkit-text-fill-color: transparent;
                                text-align: center;
                            }

                            p {
                                color: var(--blue);
                                text-transform: uppercase;
                                font-weight: 900;
                                letter-spacing: 5rem;
                                text-align: center;
                                -webkit-filter: drop-shadow(5rem -2.5rem 0 var(--pink));
                                filter: drop-shadow(5rem -2.5rem 0 var(--pink));
                            }
                        }

                        .textBlock {
                            width: 60%;

                            p {
                                text-align: left;
                            }

                            @media (${desktopMediaQuery}) {
                                width: 80%;

                                p {
                                    text-align: center;
                                }
                            }

                            @media (${tabletMediaQuery}) {
                                width: 100%;
                            }
                        }
                    }
                `}
            </style>
        </>
    );
};
