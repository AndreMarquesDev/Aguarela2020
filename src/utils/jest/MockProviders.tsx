/* eslint-disable react/jsx-no-constructed-context-values */
import React, { ReactNode } from 'react';
import { NavLinksContext } from '../../components/context/NavLinksContext';
import { TextsContext } from '../../components/context/TextsContext';
import { Locale } from '../../types/Locale';
import { textsEn, textsPt } from '../texts';

interface MockProvidersProps {
    language?: Locale;
    isMenuOpen?: boolean;
    toggleMenu?: jest.Mock;
    setNavHeight?: jest.Mock;
    children: ReactNode;
}

export const MockProviders = ({
    language = Locale.Pt,
    isMenuOpen = false,
    toggleMenu = jest.fn(),
    setNavHeight = undefined,
    children,
}: MockProvidersProps): JSX.Element => {
    return (
        <TextsContext.Provider
            value={{
                texts: language === Locale.Pt ? textsPt : textsEn,
            }}
        >
            <NavLinksContext.Provider value={{ isMenuOpen, toggleMenu, setNavHeight }}>
                {children}
            </NavLinksContext.Provider>
        </TextsContext.Provider>
    );
};
