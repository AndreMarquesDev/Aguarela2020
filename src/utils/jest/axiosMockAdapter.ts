import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { contactFormBackendUrl } from '../urls';

export const initializeAxiosMockAdapter = (delay = 0, isSuccess = true): void => {
    const mock = new MockAdapter(axios, {
        delayResponse: delay,
    });

    const httpStatusCode = isSuccess ? 200 : 500;

    mock.onPost(contactFormBackendUrl).reply(httpStatusCode, {});
};
