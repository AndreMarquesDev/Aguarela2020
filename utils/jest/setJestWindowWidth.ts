import { Breakpoint } from '../useWindowSize';

export const setJestWindowWidth = (breakpoint: Breakpoint): void => {
    let width;

    if (breakpoint === Breakpoint.Mobile) {
        width = Breakpoint.Mobile - 1;
    } else if (breakpoint === Breakpoint.Phablet) {
        width = Breakpoint.Phablet - 1;
    } else {
        width = Breakpoint.Desktop + 1;
    }

    Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: width,
    });
};
