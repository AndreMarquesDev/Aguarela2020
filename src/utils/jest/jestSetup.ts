// eslint-disable-next-line import/no-extraneous-dependencies
import mockRouter from 'next-router-mock';
import { Breakpoint } from '../useWindowSize';
import { setJestWindowWidth } from './setJestWindowWidth';

Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
    })),
});

setJestWindowWidth(Breakpoint.Desktop);

// eslint-disable-next-line import/no-extraneous-dependencies, ts/no-require-imports
jest.mock('next/router', () => require('next-router-mock'));

mockRouter.query = {
    language: 'pt',
};
