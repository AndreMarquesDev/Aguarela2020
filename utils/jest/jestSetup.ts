import { Breakpoint } from '../useWindowSize';
import { setJestWindowWidth } from './setJestWindowWidth';

Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
    })),
});

setJestWindowWidth(Breakpoint.Desktop);
