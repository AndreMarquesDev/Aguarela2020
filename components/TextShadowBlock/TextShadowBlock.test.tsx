import '@testing-library/jest-dom';
import React from 'react';
import { render, screen, RenderResult } from '@testing-library/react';
import TextShadowBlock, {
    buttonText,
    mailtoValue,
    textBlock1DataTestId,
    TextShadowBlockProps,
} from './TextShadowBlock';
import { setJestWindowWidth } from '../../utils/jest/setJestWindowWidth';

const baseProps: TextShadowBlockProps = {
    title1: 'Bem',
    title2: 'Vin',
    title3: 'Dos',
    titleMobile: 'Bem-VinDos',
    textBlock1:
        'Gestão de redes sociais está no meu ADN. Sou uma freelancer, baseada em Lisboa, com experiência em gestão de redes sociais e criação de conteúdo. A <a href="https://www.instagram.com/aguareladigital" target="_blank" rel="noreferrer" class="link">@Aguarela</a> surgiu com o objetivo de ajudar negócios locais e pequenas empresas a posicionarem-se no digital, gerar awareness, criar uma relação com o target e impulsionar vendas.',
    textBlock2:
        'Do ínicio ao fim, acompanho, monitorizo e otimizo todo o processo e estratégia de gestão de redes sociais.',
    textBlock3:
        'Através do planeamento estratégico, criação de conteúdo e implementação estou disponível para ser o vosso braço direito.',
    hasButton: false,
};

const renderComponent = (newProps?: Partial<TextShadowBlockProps>): RenderResult => {
    return render(<TextShadowBlock {...baseProps} {...newProps} />);
};

describe('<TextShadowBlock />', () => {
    test('renders properly', () => {
        renderComponent();

        const title1 = screen.getByText(baseProps.title1);
        const title2 = screen.getByText(baseProps.title2);
        const title3 = screen.getByText(baseProps.title3);
        const textBlock2 = screen.getByText(baseProps.textBlock2);
        const textBlock3 = screen.getByText(baseProps.textBlock3);

        const textBlock1 = screen.getByTestId(textBlock1DataTestId);
        const actualTextBlock1Text =
            'Gestão de redes sociais está no meu ADN. Sou uma freelancer, baseada em Lisboa, com experiência em gestão de redes sociais e criação de conteúdo. A @Aguarela surgiu com o objetivo de ajudar negócios locais e pequenas empresas a posicionarem-se no digital, gerar awareness, criar uma relação com o target e impulsionar vendas';

        const titleMobile = screen.queryByText(baseProps.titleMobile);

        expect(title1).toBeInTheDocument();
        expect(title2).toBeInTheDocument();
        expect(title3).toBeInTheDocument();
        expect(textBlock2).toBeInTheDocument();
        expect(textBlock3).toBeInTheDocument();

        expect(textBlock1).toHaveTextContent(actualTextBlock1Text);

        expect(titleMobile).not.toBeInTheDocument();
    });

    test('renders properly on mobile', () => {
        setJestWindowWidth(true);

        renderComponent();

        const titleMobile = screen.getByText(baseProps.titleMobile);

        const title1 = screen.queryByText(baseProps.title1);
        const title2 = screen.queryByText(baseProps.title2);
        const title3 = screen.queryByText(baseProps.title3);

        expect(titleMobile).toBeInTheDocument();

        expect(title1).not.toBeInTheDocument();
        expect(title2).not.toBeInTheDocument();
        expect(title3).not.toBeInTheDocument();
    });

    test('renders properly with a button', () => {
        setJestWindowWidth();

        const { container } = renderComponent({
            hasButton: true,
        });

        const buttonContent = screen.getByText(buttonText);

        expect(buttonContent).toBeInTheDocument();
        expect(buttonContent).toHaveAttribute('href', mailtoValue);

        expect(container).toMatchSnapshot();
    });
});
