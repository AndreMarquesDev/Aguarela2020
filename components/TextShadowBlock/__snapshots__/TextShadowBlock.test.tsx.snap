// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextShadowBlock /> renders properly with a button 1`] = `
<div>
  <section
    class="jsx-2290377079 container"
    data-testid="dataTestId"
  >
    <div
      class="jsx-2290377079 wrapper genericMargins"
    >
      <div
        class="jsx-2290377079 titleBlock"
      >
        <strong
          class="jsx-2290377079"
        >
          Bem
        </strong>
        <strong
          class="jsx-2290377079"
        >
          Vin
        </strong>
        <strong
          class="jsx-2290377079"
        >
          Dos
        </strong>
      </div>
      <div
        class="jsx-2290377079 textBlock"
      >
        <p
          class="jsx-2290377079 bodyText bodyText--white"
          data-testid="textShadowBlock_textBlock1"
        >
          Gestão de redes sociais está no meu ADN. Sou uma freelancer, baseada em Lisboa, com experiência em gestão de redes sociais e criação de conteúdo. A 
          <a
            class="link"
            href="https://www.instagram.com/aguareladigital"
            rel="noreferrer"
            target="_blank"
          >
            @Aguarela
          </a>
           surgiu com o objetivo de ajudar negócios locais e pequenas empresas a posicionarem-se no digital, gerar awareness, criar uma relação com o target e impulsionar vendas.
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Do ínicio ao fim, acompanho, monitorizo e otimizo todo o processo e estratégia de gestão de redes sociais.
        </p>
        <p
          class="jsx-2290377079 bodyText bodyText--white"
        >
          Através do planeamento estratégico, criação de conteúdo e implementação estou disponível para ser o vosso braço direito.
        </p>
        <div
          class="jsx-781296904 buttonWrapper"
        >
          <div
            class="jsx-781296904 buttonBackground"
          >
            <a
              class="jsx-781296904 button uppercase"
              href="mailto:catarina@aguareladigital.com"
              rel="noreferrer"
              target="_blank"
            >
              catarina@aguareladigital.com
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
